### DISCUSSION TITLE
[Question][Hw3] What should the function 'run' output?

### DISCUSSION THREAD

**Discussion by kws1207** [prosyslab-classroom/cs348-information-security#525] [Q&A]
I used iter_functions, iter_blocks, iter_instrs and set_inst_debug_location to insert my own instructions.

However, running 'make' produces an error:

File "src/main.ml", line 11, characters 5-34:
11 |   |> Llvm.print_module output_name
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This expression has type Llvm.llmodule -> unit
       but an expression was expected of type unit -> 'a
       Hint: Did you forget to wrap the expression using `fun () ->'?

 What should the function 'run' output?

**Comment by RiceBiscuits** [comment]
You can find the output type of run function at the main function.
Hint: result of the Instrument.run function will be taken by second argument of Llvm.print_module function.

**Comment by kws1207** [comment]
Thank you.
